Orbit.h

```diff
@@ -1,122 +1,104 @@
-//----------------------------------------
-// Orbit Profiler
-// Copyright Pierric Gimmig 2013-2017
-//----------------------------------------
-#pragma once
-
-//-----------------------------------------------------------------------------
-// Orbit API
-//
-// Simply include this header in your project, there is no lib and no cpp file.
-//
-// MANUAL INSTRUMENTATION:
-// The main feature of Orbit is its ability to dynamically instrument functions
-// without having to recompile or even relaunch your application.  However,
-// if you still want to manually instrument your code, you can.
-//
-// Use ORBIT_SCOPE or ORBIT_START/ORBIT_STOP *macros* only.
-// DO NOT use OrbitScope(...)/OrbitStart(...)/OrbitStop() directly.
-// NOTE: You need to use string literals.  Dynamic strings are not supported
-// yet.
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORBIT_H_
+#define ORBIT_H_
+
+#include <stdint.h>
+
+// Orbit API (header-only)
 //
-// DLL LOADING:
-// If you want to control the loading of Orbit64.dll, i.e. you don't want to
-// have to manually inject it into your application, use the OrbitAPI class.
-// Calling OrbitAPI::Connect(...) will load the dll and try to connect to the
-// Orbit instance specified in parameters.  Note that Orbit64.dll needs to be
-// next to your exectutable.  You can also change the code below
-// (OrbitAPI::Init()).
-//-----------------------------------------------------------------------------
-
-#define ORBIT_SCOPE(name) OrbitScope ORBIT_UNIQUE(ORB)(ORBIT_LITERAL(name))
-#define ORBIT_START(name) OrbitStart(ORBIT_LITERAL(name))
-#define ORBIT_STOP OrbitStop()
-#define ORBIT_SEND(ptr, num) OrbitSendData(ptr, num)
-
-//-----------------------------------------------------------------------------
-struct OrbitAPI {
-  static inline bool Connect(const char* a_Host, int a_Port = 1789);
-  static inline bool IsConnected();
-
- private:
-  static inline void Init();
-
-  typedef void (*InitRemote)(char*);
-  typedef bool (*GetBool)();
-
-  static inline HINSTANCE& GetHandle();
-  static inline InitRemote& GetInitRemote();
-  static inline GetBool& GetIsConnected();
-};
+// While the main feature of Orbit is its ability to dynamically instrument
+// functions, manual instrumentation is still possible using the macros below.
+// These macros call empty functions that Orbit dynamically instruments.
```

Comment:
Probably not necessary right now, but examples would be really nice and would make it a lot quicker to pick this up and use.

Reply:
Yes totally agree. Will add a proper "howto" either directly in the .h or in the upcoming docs.